db/sequence_allocator.go

```diff
@@ -185,6 +179,18 @@ func (s *sequenceAllocator) nextSequence(ctx context.Context) (sequence uint64,
 	return sequence, nil
 }
 
+func (s *sequenceAllocator) _nextSequence(ctx context.Context) (sequence uint64, sequencesReserved bool, err error) {
+	if s.last >= s.max {
+		if err := s._reserveSequenceRange(ctx); err != nil {
```

Comment:
Have a way to reserve a sequence range from current sequence up to the high doc.Sequence